### PR TITLE
Fix several issues reported by ESLint

### PR DESCRIPTION
**Pull request by imobachgs**
In order to build our RPM package, we need to tackle the issues that were reported by ESLint. Especially interesting is the callbacks dependencies problem (see 27d2d81).

## useCallbacks

Every time a component is rendered, the functions which are defined inside are created again (so each time it is a new function). That's perfectly fine. The problem appears when a function `myFunc` appears as a dependency in a `useEffect` call, as it will cause the `useEffect` function to run again and again (remember, the `myFunc` element is different on every render). That's where `useCallback` comes into play.

* [useCallback documentation](https://reactjs.org/docs/hooks-reference.html#usecallback)
* Should we use `useCallback` for every function? No, please. See [Your Guide to React.useCallback()](https://dmitripavlutin.com/dont-overuse-react-usecallback/) by Dmitri Pavlutin.

## Defining Auxiliary Functions Inside or Outside the Components

At this point, we have been defining most of the auxiliary functions within the components. That's fine for those functions that are using closures and, somehow, need to be re-created on each render. However, we could move other functions outside of the component (so they are not recreated if not needed).

Generally speaking, you could move any function outside of the component (passing all needed pieces). But if you are not going to reuse such a function, I would say that it does not worth. Let's consider the following examples (check the comments, please).

```js
function initialN(max) {
    return parseInt(Math.random() * max);
}

function Counter({ max = 100 }) {
    // initialN just depends on its argument. No need to re-create it again and again.
    const [n, setN] = useState(initialN(max));

    // In order to move this function out, I would need to create another one an
    // assign it to the onClick attribure:
    //
    // onClick={() => increment(n, setN)}
    //
    // So it does not worh.
    const increment = () => setN(n+1);

    return (
            <>
                <div>{n}</div>
                <button onClick={increment}>Increment</button>
            </>
    );
}

export default Counter;
```

### CODE REVIEW

File: src/lib/NetworkClient.js [hunk 1]
```diff
@@ -121,14 +121,16 @@ class NetworkClient {
...
      * @param {Interface} Interface - Obtains the list of networks available for the given interface
-     * @returns {Promise<String|Error>}
+     * @returns {Promise<Array<String>|Error>}
      */
-    getEssidList(iface) {
+    async getEssidList(iface) {
         const link_up = `ip link set ${iface} up`;
         const scan = `iwlist ${iface} scan`;
         const grep_and_cut_essid = "/usr/bin/grep ESSID | /usr/bin/cut -d':' -f2 | /usr/bin/cut -d'\"' -f2";
-        const sort = "/usr/bin/sort -u";
```
Comment (teclator) [inline src/lib/NetworkClient.js:130]: :+1:

File: src/components/WirelessForm.js [hunk 2]
```diff
@@ -60,20 +61,19 @@ const WirelessForm = ({ isOpen, onClose, iface, connection }) => {
...
-    const refreshList = (name) => {
-        client.getEssidList(name)
-                .then((result) => {
-                    essid
-                        ? setEssidList([...new Set([...result.split("\n"), essid])])
-                        : setEssidList([...new Set([...result.split("\n")])]);
+    const refreshList = useCallback((name) => {
+        fetchEssidList(name)
+                .then(result => {
+                    const list = (oldEssid && !result.includes(oldEssid)) ? [...result, oldEssid] : [...result];
```
Comment (teclator) [inline src/components/WirelessForm.js:67]: :+1:

**Review by teclator — APPROVED** [review]
LGTM

**Review by dgdavid — APPROVED** [review]
LGTM, thanks a lot @imobachgs

**Comment by imobachgs** [comment]
@dgdavid @teclator Thanks for the review!

**Pull request merged**